### ISSUE TITLE
[Bug]: Bug in MO Startup Process

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#14092]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main,1.1-dev,1.0-dev

### Commit ID

4541bf1364e3c720c929016d16df39150569a0d2

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

I've identified a critical issue in the startup process of the MO that needs attention. MO begins listening on port 6001 before the system has fully completed its bootstrap process. This could potentially lead to a series of unknown and undesired issues.

<img width="1719" alt="企业微信截图_deb54cf0-25f6-42ff-91d7-6500957ff021" src="https://github.com/matrixorigin/matrixone/assets/77312370/77f44ac6-67e4-424c-bbf3-e3307e2ec783">

### Expected Behavior

Ensure that the server port(default: 6001) is not listened on until the system has fully bootstraped. 

### Steps to Reproduce

```Markdown
start mo locally
```

**Comment by daviszhen** [comment]
@sukki37 这地方有一个点。整个集群的启动是由mo_task 调度任务驱动的。某个cn会执行集群初始化任务。mo_task 需要登录mo，取system init 任务初始化sys租户。目前的集群初始化逻辑需要mo 先听6001端口。同样metric/trace好像也有类似的逻辑。但是站在mo端，无法区分连接进来是初始化mo的还是普通用户的查询

**Comment by daviszhen** [comment]
未投入

**Comment by daviszhen** [comment]
未投入

**Comment by sukki37** [comment]
The fix for this bug depends on the refactoring of the framework upgrade #13633 .

**Comment by qingxinhome** [comment]
Waiting for the new upgrade framework to resolve

**Comment by qingxinhome** [comment]
PR has been submitted

**Comment by qingxinhome** [comment]
PR has been submitted

**Comment by qingxinhome** [comment]
The issue has been resolved and PR has been submitted. Please verify and test it @sukki37 @aressu1985 @daviszhen

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
fixed:275af9cc9a4d113bd5ddcb5381eb9a9d15e0dd87
